nstar/src/main.rs

```diff
@@ -292,19 +292,28 @@ fn stop<'a, S: Read + Write, I: Iterator<Item = &'a str>>(
     let pattern = cmd.next().unwrap_or(".*");
     let re = regex::Regex::new(pattern).context("Invalid regex")?;
     if let Response::Containers(containers) = run(stream, Request::Containers)? {
-        for container in containers
+        let running_container_names: Vec<String> = containers
             .iter()
             .filter(|ref c| c.process.is_some()) // Filter not running containers
             .filter(|ref c| re.is_match(&c.manifest.name))
-        {
-            let response = run(stream, Request::Stop(container.manifest.name.clone()))?;
-            println!(
-                "Stopped {}:{}: {:?}",
-                container.manifest.name, container.manifest.version, response
-            );
+            .map(|ref c| c.manifest.name.clone())
+            .collect();
+
+        if running_container_names.is_empty() {
+            Err(anyhow!(
+                "No running container match the pattern: {}",
+                pattern
+            ))
```

Comment:
Not sure if this is really an error. If there's nothing to do - there's nothing to do.